core/kernel/classes/class.Class.php

```diff
@@ -430,7 +434,9 @@ public function deleteInstances($resources, $deleteReference = false)
      */
     public function delete($deleteReference = false)
     {
-        return (bool) $this->getImplementation()->delete($this, $deleteReference);
+        $delete = (bool) $this->getImplementation()->delete($this, $deleteReference);
+        $this->getEventManager()->trigger(new ClassDeletedEvent($this));
```

Comment:
it would be triggered in any case
even If `getImplementation()->delete(` fail / return false
does the triggering should depend on it?

Reply:
I trigger an event after an event.

Reply:
I don't know much about your task,
but I see that you added your code which triggers the event right after some older code which removes a class from the persistence and returns status of this operation (I assume **false** if failed to delete)
So my question: is it right to trigger ClassDeletedEvent even if deletion from persistence failed?

Reply:
I think what @pnal means is we should have a check:

```
$delete = (bool) $this->getImplementation()->delete($this, $deleteReference);

if ($delete) {
    $this->getEventManager()->trigger(new ClassDeletedEvent($this));
}
```